flake8_trio/visitors/helpers.py

```diff
@@ -125,6 +142,62 @@ def iter_guaranteed_once(iterable: ast.expr) -> bool:
     return False
 
 
+def cst_literal_eval(node: cst.BaseExpression) -> Any:
+    ast_node = cst.Module([cst.SimpleStatementLine([cst.Expr(node)])]).code
+    try:
+        return ast.literal_eval(ast_node)
+    except Exception:  # noqa: PIE786
+        return None
+
+
+def iter_guaranteed_once_cst(iterable: cst.BaseExpression) -> bool:
```

Comment:
ugly, but works pretty much like the ast version.

Reply:
you mean using `iter_guaranteed_once` ?
It has all the same tests that the ast version had, and vast majority of the logic is the same, so I'm pretty confident it's equivalent actually. And if/when 103/104 is converted to libcst there will be no ast users left using this.